plugins/woocommerce-admin/client/products/test/product-form-actions.spec.tsx

```diff
@@ -417,6 +433,52 @@ describe( 'ProductFormActions', () => {
 			);
 		} );
 	} );
+
+	describe( 'when the form is invalid', () => {
+		[ 'Save draft', 'Preview', 'Publish' ].forEach( ( buttonText ) => {
+			it( `should have the ${ buttonText } button disabled`, () => {
+				render(
+					<Form initialValues={ {} } validate={ validate }>
+						<ProductFormActions />
+					</Form>
+				);
+				const actionButton = screen.getByText( buttonText );
+				expect( actionButton ).toBeDisabled();
+			} );
+		} );
+
+		it( 'should have the Publish options menu button disabled', () => {
+			render(
+				<Form initialValues={ {} } validate={ validate }>
+					<ProductFormActions />
+				</Form>
+			);
+			expect( screen.getByLabelText( 'Publish options' ) ).toBeDisabled();
+		} );
+
+		it( 'should have the Publish options menu items disabled', () => {
+			render(
+				<Form initialValues={ { id: 1 } } validate={ validate }>
+					<ProductFormActions />
+				</Form>
+			);
+
+			const publishOptionsButton =
+				screen.getByLabelText( 'Publish options' );
+
+			userEvent.click( publishOptionsButton );
+
+			const optionsMenu = screen.getByRole( 'menu' );
+			const menuItems = within( optionsMenu ).getAllByRole( 'menuitem' );
+			// Verify only the first two items (Publish & duplicate|Update & duplicate
+			// and Copy to a new draft) from the action menu.
+			const firstTwoItems = menuItems.slice( 0, 2 );
```

Comment:
Instead of depending on the position of the menu items, can you rewrite this to use `screen.getByText()` as you have in the other tests? It would make this test more robust, in case the menu order changes in the future.

Reply:
The reason I did it that way was because the text of the first item is dynamic. So if I change this test according your suggestion that will end up testing the text/change behavior which is not part of this PR. But I'll give it a try considering we can force the text of the first item to have a particular value at the moment the test is running. Also it's a very good suggestion.

Reply:
Thanks for updating this!